pkg/api/providers/cohere/api.go

```diff
@@ -0,0 +1,25 @@
+package cohere
+
+import (
+	"fmt"
+	"net/http"
+
+	"glide/pkg/api/providers"
+)
+
+// CohereAPIConfig represents the API configuration for Cohere
+func CohereApiConfig(APIKey string) providers.ProviderApiConfig {
```

Comment:
I would really go a different route while organising this LLM logic. 

It's more trivial to have these specifics incorporated into clients. So we have a client class per each LLM.

That client class knows how to send the request (e.g. that the base URL, in what format the LLM accepts requests, where API token should be provided, where is different API like chat API located, etc).

Now those clients could have different API e.g. methods and their params or could be unified right away.

Then, if you zoom out, provider clients are going to be grouped into a provider pool. On that level, the resiliency/load balancing logic happens.

If you zoom out even further, Glide API would receive a request and pass it further to a provider pool and then it would respond back. 

Visually, it looks like this:

![Untitled-2023-12-16-1924](https://github.com/modelgateway/Glide/assets/9402690/724abfec-5bcf-47f1-a85e-c99b0a5f75ba)

Because of all of this, I think Glide Unified API is one of the most important things to think about as of now.